src/bin/playbot.rs

```diff
@@ -250,9 +253,17 @@ fn main() {
                 shorten_key: bitly_key.clone(),
                 cache: cache.clone(),
             };
-            threads.push(thread::spawn(move || {
-                bot.main_loop();
-            }));
+            threads.push(thread::Builder::new()
+                                         .name(format!("{}@{}", nick, server_addr))
+                                         .spawn(move || {
+                if let Err(e) = catch_unwind(AssertUnwindSafe(|| bot.main_loop())) {
```

Comment:
That makes rustc complain about a *lot* of unsatisfied `RefUnwindSafe` bounds somewhere in the `irc` crate.